src/main/java/org/springframework/cli/command/recipe/RecipeParameterParser.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cli.command.recipe;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cli.command.recipe.catalog.RecipeParameterDefinitions;
+
+
+/**
+ * Provides utility methods for formatting and parsing deployment properties.
+ *
+ * @author Eric Bottard
+ * @author Mark Fisher
+ * @author Janne Valkealahti
+ * @author Christian Tzolov
+ * @author Gunnar Hillert
+ * @author Ilayaperumal Gopinathan
+ * @author Glenn Renfro
+ */
+public final class RecipeParameterParser {
```

Comment:
I don't think we should open this rabbit hole and introduce complex parameter types.

Reply:
It only reads String parameters, for now, using the class from SCDF.
I left the API using this complex `Map<String,List<String>>` data structure that I created beforehand.
I'd suggest we discuss this with @markpollack and leave it as is for now. 
I could then either remove or refactor it in a new PR?